mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -256,22 +257,26 @@ public MailboxId rename(Mailbox mailbox) throws MailboxException {
     }
 
     @Override
-    public List<Mailbox> list() throws MailboxException {
-       File maildirRoot = maildirStore.getMaildirRoot();
-       List<Mailbox> mailboxList = new ArrayList<>();
+    public Flux<Mailbox> list() {
+        try {
+            File maildirRoot = maildirStore.getMaildirRoot();
+            List<Mailbox> mailboxList = new ArrayList<>();
 
-       if (maildirStore.getMaildirLocation().endsWith("/" + MaildirStore.PATH_DOMAIN + "/" + MaildirStore.PATH_USER)) {
-           File[] domains = maildirRoot.listFiles();
-           for (File domain : domains) {
-               File[] users = domain.listFiles();
-               visitUsersForMailboxList(domain, users, mailboxList);
-           }
-           return mailboxList;
-       }
+            if (maildirStore.getMaildirLocation().endsWith("/" + MaildirStore.PATH_DOMAIN + "/" + MaildirStore.PATH_USER)) {
+                File[] domains = maildirRoot.listFiles();
+                for (File domain : domains) {
+                    File[] users = domain.listFiles();
+                    visitUsersForMailboxList(domain, users, mailboxList);
```

Comment:
Shouldn't `visitUsersForMailboxList` operation be reactified?

Reply:
I saw some work done on that here https://github.com/linagora/james-project/pull/3281/commits/ca0bed839c20b4f0d76b025d2546b8417cb3893c